llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -1331,7 +1331,7 @@ struct AAPointerInfoImpl
         // itself either.
         bool Inserted = ExclusionSet.insert(&I).second;
 
-        if (!FnReachabilityAA ||
```

Comment:
We don't need to insert/remove if the Reachability AA is not present. Simply skip the entire thing.